### PR TITLE
[cb, cb_explore] small refactor for reuse

### PR DESCRIPTION
**Pull request by lalo**
Prompted by https://github.com/VowpalWabbit/vowpal_wabbit/pull/2680

### CODE REVIEW

File: vowpalwabbit/cb_algs.cc [hunk 1]
```diff
@@ -100,6 +100,11 @@ void output_example(vw& all, cb& data, example& ec, CB::label& ld)
   cb_to_cs& c = data.cbcs;
   if (!CB::is_test_label(ld)) loss = get_cost_estimate(c.known_cost, c.pred_scores, ec.pred.multiclass);
 
+  generic_output_example(all, loss, ec, ld);
+}
+
+void generic_output_example(vw& all, float loss, example& ec, CB::label& ld)
```
Comment (jackgerrits) [inline vowpalwabbit/cb_algs.cc:106]: Can ld be const?

File: vowpalwabbit/cb_algs.h [hunk 2]
```diff
@@ -49,19 +49,28 @@ float get_cost_pred(
...
 
+// doubly robust estimate
 inline float get_cost_estimate(const CB::cb_class& observation, const COST_SENSITIVE::label& scores, uint32_t action)
 {
   for (auto& cl : scores.costs)
     if (cl.class_index == action) return get_cost_estimate(observation, action, cl.x) + cl.x;
   return get_cost_estimate(observation, action);
 }
 
+inline float get_cost_estimate(CB::label& ld, uint32_t action)
```
Comment (jackgerrits) [inline vowpalwabbit/cb_algs.h:67]: Can ld be const?

File: vowpalwabbit/cb_explore.cc [hunk 3]
```diff
@@ -253,7 +254,7 @@ void print_update_cb_explore(vw& all, bool is_test, example& ec, std::stringstre
   }
 }
 
-void output_example(vw& all, cb_explore& data, example& ec, CB::label& ld)
+float calc_loss(cb_explore& data, example& ec, CB::label& ld)
```
Comment (jackgerrits) [inline vowpalwabbit/cb_explore.cc:257]: Can ld be const?

**Review by jackgerrits — APPROVED** [review]

**Review by jackgerrits — APPROVED** [review]

File: vowpalwabbit/cb_algs.cc (hunk 1 above)
Comment (lalo) [reply vowpalwabbit/cb_algs.cc]: done

File: vowpalwabbit/cb_algs.h (hunk 2 above)
Comment (lalo) [reply vowpalwabbit/cb_algs.h]: done

File: vowpalwabbit/cb_explore.cc (hunk 3 above)
Comment (lalo) [reply vowpalwabbit/cb_explore.cc]: done

**Pull request merged**